app/Modules/Availability/AvailabilityService.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+namespace App\Modules\Availability;
+
+use App\Exceptions\OverlappingAvailabilityException;
+use App\Exceptions\PrivilegeException;
+use App\Modules\Availability\AvailabilityInPastException;
+use App\User;
+use Carbon\Carbon;
+
+class AvailabilityService
+{
+    /** @var mixed App\User */
+    protected $user;
+
+    /** HIS MIN TIME BLOCK TIME AVAILABILITY SHOULD BE */
+    const MIN_BLOCK_TIME = 30;
+
+    public function __construct(User $user)
+    {
+        $this->user = $user;
+    }
+
+    /**
+     * Validate that two given times are valid within the context of other availability.
+     *
+     * @param  Carbon  $from
+     * @param  Carbon  $to
+     * @param int $doNotCheck - Pass in an int to exclude from the search
+     * @param  User  $user - User
+     * @param  int|null  $excluded  - Availability ID to be excluded from the check.
+     * @throws AvailabilityInPastException - Cannot add availability that occurs in the past
+     * @return bool
+     */
+    public function validateAvailabilityTimes(Carbon $from, Carbon $to, User $user, int $doNotCheck =  NULL): bool
+    {
+        if ($from->isAfter($to)) {
+            return false;
+        }
+
+        if ($from->isBefore(new Carbon())) {
```

Comment:
Could be worth using https://laravel.com/docs/master/helpers#method-throw-if

Reply:
done